packages/server/src/analytics/storage.ts

```diff
@@ -0,0 +1,100 @@
+import { FirestoreService } from '../service/firestore'
+import { analyticsLogic } from './logic'
+
+const firestore = new FirestoreService()
+
+// Used to save the document
+export interface AnalyticsStorageSchema {
+  readonly id: 'onlyOne'
+  lastQueryTime: number
+  totalSignups: number
+  yesterdayDate: number
+  yesterdaySignups: number
+}
+
+class AnalyticsStorage {
+  private synced = false
+
+  private get doc() {
+    return firestore.db.collection('Analytics').doc('storage')
+  }
+
+  private storage: AnalyticsStorageSchema = {
+    id: 'onlyOne',
+    lastQueryTime: 0,
+    totalSignups: 0,
+    yesterdayDate: 0,
+    yesterdaySignups: 0,
+  }
+
+  /**
+   * Refreshes the in-memory values or creates the default document in
+   * the collection.
+   *
+   * If this function is successful, the DB is considered as synchronized
+   * for 1 minute.
+   */
+  initializeOrSync = async () => {
+    const get = await this.doc.get()
+    const data = get.data()
+
+    if (!data) {
+      await this.doc.set(this.storage, { merge: true })
+    } else {
+      this.storage = {
+        ...data as AnalyticsStorageSchema,
+        id: 'onlyOne',
+      }
+    }
+
+    // We need this timer here since we use `AnalyticsStorage` at all files
```

Comment:
I don't understand why this is necessary.  You only call it `initializeOrSync` once.  You should remove this.